lib/graphql/schema/member/has_deprecation_reason.rb

```diff
@@ -16,6 +15,10 @@ def deprecation_reason=(text)
           if text.nil?
             remove_directive(GraphQL::Schema::Directive::Deprecated)
           else
+            if defined?(@deprecation_reason) && @deprecation_reason
+              remove_directive(GraphQL::Schema::Directive::Deprecated)
+            end
+            @deprecation_reason = text
```

Comment:
Should this line always be run? If `text` is `nil`, we would want to reset `@deprecation_reason` in case of future `deprecated_reason` usage?

Reply:
Ahh, good catch! Thanks for taking a look, I pushed a test for this case and a fix in 682e94f75